src/solver/conv_cellfft.cpp

```diff
@@ -0,0 +1,603 @@
+    const size_t bdx   = p.id == 0 ? 256 : 512;
+    const size_t gdx   = (p.n + 15) >> 4;
+    const std::vector<size_t> block{bdx, 1, 1};
+    const std::vector<size_t> grid{gdx * bdx, p.k, 1};
+    std::ostringstream options;
+    GenerateClangDefsym(options, "ROCM_METADATA_VERSION", ctx.rmv.UseV3() ? 5 : 4);
+    return KernelInfo{options.str(), block, grid, file_name, g_knames[kid]};
+}
+static KernelInfo get_kernel_c2r(const ConvolutionContext& ctx,
+                                 const cellfft_param_t& p,
+                                 const std::string& file_name,
+                                 uint32_t relu)
+{
+    const uint32_t shx = 4 - p.id;
+    const uint32_t kid = START_C2R + (p.id << 5) + (relu << (p.id + 4)) + p.cny - 1;
+    const size_t gdx   = (p.m + (1 << shx) - 1) >> shx;
+    const std::vector<size_t> block{256, 1, 1};
+    const std::vector<size_t> grid{gdx << 8, p.n, 1};
+    std::ostringstream options;
+    GenerateClangDefsym(options, "ROCM_METADATA_VERSION", ctx.rmv.UseV3() ? 5 : 4);
+    return KernelInfo{options.str(), block, grid, file_name, g_knames[kid]};
+}
+static KernelInfo get_kernel_c2r_grid(const ConvolutionContext& ctx,
+                                      const cellfft_param_t& p,
+                                      const std::string& file_name,
+                                      uint32_t relu)
+{
+    const uint32_t shx = 4 - p.id;
+    const uint32_t kid = START_C2R_X + ((p.id << 1) | relu);
+    const size_t gdx   = (p.m + (1 << shx) - 1) >> shx;
+    const std::vector<size_t> block{256, 1, 1};
+    const std::vector<size_t> grid{gdx << 8, p.n, 1};
+    std::ostringstream options;
+    GenerateClangDefsym(options, "ROCM_METADATA_VERSION", ctx.rmv.UseV3() ? 5 : 4);
+    return KernelInfo{options.str(), block, grid, file_name, g_knames[kid]};
+}
+static KernelInfo get_kernel_c2r_grad(const ConvolutionContext& ctx,
+                                      const cellfft_param_t& p,
+                                      const std::string& file_name)
+{
+    uint32_t nmax = p.cnx > p.cny ? p.cnx : p.cny;
+    uint32_t nmin = p.cnx > p.cny ? p.cny : p.cnx;
+    uint32_t shx  = 4 - p.id;
+    uint32_t kid  = START_C2R + (p.id << 5) + p.cny - 1;
+    bool cc0      = (p.cnx == p.cny) && ((p.cnx == 3) || (p.cnx == 5) || (p.cnx == 7));
+    bool cc1      = (nmin == 1) && ((nmax & 1) != 0) && (nmax > 1) && (nmax <= 9);
+    if(cc0 || cc1)
+    {
+        kid = START_C2R_S + 11 * p.id + (cc0 ? 8 : (p.cnx > p.cny ? 0 : 4)) + (nmax >> 1) - 1;
+    }
+    size_t gdx = (p.m + (1 << shx) - 1) >> shx;
+    const std::vector<size_t> block{256, 1, 1};
+    const std::vector<size_t> grid{gdx << 8, p.n, 1};
+    std::ostringstream options;
+    GenerateClangDefsym(options, "ROCM_METADATA_VERSION", ctx.rmv.UseV3() ? 5 : 4);
+    return KernelInfo{options.str(), block, grid, file_name, g_knames[kid]};
+}
+static void get_solution(const ConvolutionContext& ctx, ConvSolution& sol, const cellfft_param_t& p)
```

Comment:
I'd suggest renaming it to something like `fill_kernel_build_parameters`. Or `FillKernelBuildParameters`. Most of the library functions are camel-cased, but I don't think it is enforced.